compiler/rustc_mir_transform/src/jump_threading.rs

```diff
@@ -557,7 +571,9 @@ impl<'a, 'tcx> TOFinder<'a, 'tcx> {
                 // `SetDiscriminant` may be a no-op if the assigned variant is the untagged variant
                 // of a niche encoding. If we cannot ensure that we write to the discriminant, do
                 // nothing.
-                let Ok(enum_layout) = self.ecx.layout_of(enum_ty) else { return };
+                let Some(enum_layout) = self.ecx.layout_of(enum_ty).discard_interp_error() else {
```

Comment:
This seems really easy to mix up by writing `let Ok(_)`, as noted above.

I'm not certain the best way to avoid this, except for going way overkill and implementing some custom struct (like not a `Result` but a struct wrapper that makes the `Result` inaccessible) that implements `Try` and forces you to deal with the error by either doing `?` (via `Try`) or `.discard_interp_error`...

Reply:
Yeah that's the only type-based alternative I can think of.